model/project_ref.go

```diff
@@ -1977,6 +1981,19 @@ func SaveProjectPageForSection(projectId string, p *ProjectRef, section ProjectP
 	var err error
 	switch section {
 	case ProjectPageGeneralSection:
+		if utility.FromBoolPtr(p.RepotrackerDisabled) && p.RemotePath == "" {
+			var pRef *ProjectRef
+			pRef, err = FindBranchProjectRef(projectId)
```

Comment:
We don't want to do a query to see what's _currently_ true in the DB -- this is the update function, so we're going to overwrite whatever  currently is in the DB with whatever p.RemoteFile is. So we should just be able to error here, since this is the bad state (i.e. the repotracker is enabled and config file isn't set).